internal/brokers/broker.go

```diff
@@ -302,48 +307,47 @@ func (b Broker) validateAndMarshalUser(rawMsg json.RawMessage) (data string, err
 		}
 	}
 	if err := json.Unmarshal(rawMsg, &uInfo); err != nil {
-		return "", fmt.Errorf("message is not JSON formatted: %v", err)
+		return cache.UserInfo{}, fmt.Errorf("message is not JSON formatted: %v", err)
 	}
 
 	// Validate username
 	if uInfo.Name == "" {
-		return "", fmt.Errorf("empty username")
+		return cache.UserInfo{}, fmt.Errorf("empty username")
 	}
 
 	// Validate home and shell directories
-	for _, dir := range []string{uInfo.Dir, uInfo.Shell} {
-		if !filepath.IsAbs(filepath.Clean(dir)) {
-			return "", fmt.Errorf("%q is not an absolute path", dir)
-		}
+	if !filepath.IsAbs(filepath.Clean(uInfo.Dir)) {
+		return cache.UserInfo{}, fmt.Errorf("value provided for homedir is not an absolute path: %s", uInfo.Dir)
+	}
+	if !filepath.IsAbs(filepath.Clean(uInfo.Shell)) {
+		return cache.UserInfo{}, fmt.Errorf("value provided for shell is not an absolute path: %s", uInfo.Shell)
 	}
 
-	// Generate UID and GIDs
-	hasher := crypto.SHA256.New()
-
-	uid, err := hasher.Write([]byte(b.Name + uInfo.UUID))
-	if err != nil {
-		return "", fmt.Errorf("could not generate UID")
+	// Validate UUID and generate UID
+	if uInfo.UUID == "" {
+		return cache.UserInfo{}, fmt.Errorf("empty UUID")
 	}
-	uInfo.UID = (uid % (100000 - 65537)) + 65536 // Ensures that UID is between 65536 and 100000
+	uInfo.UID = generateID(brokerName + uInfo.UUID)
 
+	// Validate UGIDs and generate GIDs
 	for _, g := range uInfo.Groups {
-		group := cache.GroupInfo{Name: g.Name}
-
-		gid, err := hasher.Write([]byte(b.Name + g.UGID))
-		if err != nil {
-			return "", fmt.Errorf("could not generate GID")
+		if g.UGID == "" {
+			return cache.UserInfo{}, fmt.Errorf("group %q has empty UGID", g.Name)
 		}
-		group.GID = (gid % (100000 - 65537)) + 65536 // Ensures that GID is between 65536 and 100000
-
-		uInfo.UserInfo.Groups = append(uInfo.UserInfo.Groups, group)
+		gid := generateID(brokerName + g.UGID)
+		uInfo.UserInfo.Groups = append(uInfo.UserInfo.Groups, cache.GroupInfo{Name: g.Name, GID: gid})
 	}
 
-	d, err := json.Marshal(uInfo.UserInfo)
-	if err != nil {
-		return "", fmt.Errorf("can't marshal UserInfo: %v", err)
-	}
+	return uInfo.UserInfo, nil
+}
 
-	return string(d), nil
+// generatedID generates an integer number based on the provided string.
+func generateID(str string) int {
+	var sum int
+	for i, c := range str {
+		sum += int(uint(c)*uint(i+1)) - i
```

Comment:
`why -i` ? Maybe an comment is in order.

Reply:
Trying to avoid as many conflicts as possible... There could be cases where the sum matches, but subtracting the index adds another layer to avoid some LCM-like (least common multiple) cases

Reply:
just add a comment, for now :) We can revisit anyway!